test/extended/images/resolve.go

```diff
@@ -128,6 +127,83 @@ var _ = g.Describe("[Feature:ImageLookup][registry][Conformance] Image policy",
+
+		g.By("auto replacing local references on Deployments")
+		deployment, err := oc.KubeClient().AppsV1().Deployments(oc.Namespace()).Create(&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "resolve",
+				Annotations: map[string]string{
+					"alpha.image.policy.openshift.io/resolve-names": "*",
+				},
+			},
+			Spec: appsv1.DeploymentSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{"resolve": "true"},
+				},
+				Template: kapiv1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"resolve": "true"}},
+					Spec: kapiv1.PodSpec{
+						TerminationGracePeriodSeconds: &one,
+						Containers: []kapiv1.Container{
+							{Name: "resolve", Image: "busybox:latest", Command: []string{"/bin/true"}},
+						},
+					},
+				},
+			},
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(deployment.Spec.Template.Spec.Containers[0].Image).To(o.Equal(tag.Image.DockerImageReference))
+
+		g.By("auto replacing local references on DaemonSets")
+		daemonset, err := oc.KubeClient().AppsV1().DaemonSets(oc.Namespace()).Create(&appsv1.DaemonSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "resolve",
+				Annotations: map[string]string{
+					"alpha.image.policy.openshift.io/resolve-names": "*",
+				},
+			},
+			Spec: appsv1.DaemonSetSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{"resolve": "true"},
+				},
+				Template: kapiv1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"resolve": "true"}},
+					Spec: kapiv1.PodSpec{
+						TerminationGracePeriodSeconds: &one,
+						Containers: []kapiv1.Container{
+							{Name: "resolve", Image: "busybox:latest", Command: []string{"/bin/true"}},
+						},
+					},
+				},
+			},
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(daemonset.Spec.Template.Spec.Containers[0].Image).To(o.Equal(tag.Image.DockerImageReference))
+
+		g.By("auto replacing local references on StatefulSets")
+		statefulset, err := oc.KubeClient().AppsV1().StatefulSets(oc.Namespace()).Create(&appsv1.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "resolve",
+				Annotations: map[string]string{
+					"alpha.image.policy.openshift.io/resolve-names": "*",
+				},
+			},
+			Spec: appsv1.StatefulSetSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{"resolve": "true"},
+				},
+				Template: kapiv1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"resolve": "true"}},
+					Spec: kapiv1.PodSpec{
+						TerminationGracePeriodSeconds: &one,
+						Containers: []kapiv1.Container{
+							{Name: "resolve", Image: "busybox:latest", Command: []string{"/bin/true"}},
+						},
+					},
+				},
+			},
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(statefulset.Spec.Template.Spec.Containers[0].Image).To(o.Equal(tag.Image.DockerImageReference))
```

Comment:
How about also adding CronJob and job here as well?

Reply:
I'll add them too